pochivm/api_base.h

```diff
@@ -115,113 +121,161 @@ Value<T>::operator Value<U>() const
     return StaticCast<U>(*this);
 }
 
+
+// Helper for Arithmetic ops operator overloading. Returns an arithmetic expression of the form
+// lhs OP rhs where OP is the operator specified by `ExprType`
+//
+template <AstArithmeticExprType expr_type, typename T, typename U, 
+          typename = std::enable_if_t<AstTypeHelper::primitive_type_supports_arithmetic_expr_type<T, expr_type>::value>,
+          typename = std::enable_if_t<AstTypeHelper::primitive_type_supports_arithmetic_expr_type<U, expr_type>::value> >
+Value<typename AstTypeHelper::ArithReturnType<T, U>::type> DoArith(const Value<T>& lhs, const Value<U>& rhs)
```

Comment:
The 'DoArith' name is very blurry (one cannot understand or even guess what this function is doing from its name). Can you change it to a more descriptive name?